### PR TITLE
Add sample project with Maven cache configuration for various plugins

### PR DESCRIPTION
**Pull request by bigdaz**

### CODE REVIEW

File: maven-caching-sample/avro-project/pom.xml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+                      <property><name>stringType</name></property>
+                      <property><name>imports</name></property>
+
+                      <property><name>customConversions</name></property>
+                      <property><name>customLogicalTypeFactories</name></property>
+                      <property><name>templateDirectory</name></property>
+                      <property><name>velocityToolsClassesNames</name></property>
+                    </properties>
+                    <ignoredProperties>
```
Comment (runningcode) [inline maven-caching-sample/avro-project/pom.xml:106]: I think some comments on why these properties were ignored might help.

File: maven-caching-sample/pmd-project/pom.xml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                          <property><name>skipPmdError</name></property>
+                          <property><name>renderProcessingErrors</name></property>
+                          <property><name>renderRuleViolationPriority</name></property>
+                          <property><name>renderViolationsByPriority</name></property>
+                        </properties>
+                        <ignoredProperties>
+                          <ignore>showPmdLog</ignore>
+                          <ignore>reactorProjects</ignore>
+                          <ignore>session</ignore>
+                          <!-- Temporary file location can be ignored, assuming no external process is relying on these ruleset files. -->
```
Comment (runningcode) [inline maven-caching-sample/pmd-project/pom.xml:92]: 👍

**Review by runningcode — APPROVED** [review]
Looks good to me!

**Comment by etiennestuder** [comment]
@bigdaz Don't wait for my review but merge once @runningcode is happy with it. Someone from Maven team to take a look would be great, but they might not have time given the upcoming GE 2021.1 release.

**Comment by bigdaz** [comment]
@runningcode I add some more comments on why properties were ignored. This PR is now merged.

**Pull request merged**